FRAuth/FRAuth/Callbacks/FRAppIntegrityCallback.swift

```diff
@@ -117,32 +133,35 @@ public class FRAppIntegrityCallback: MultipleValuesCallback {
     /// Sets `keyId` value in callback response
     /// - Parameter keyId: Base64 String value of keyId
     public func setkeyId(_ keyId: String) {
-        self.inputValues[self.keyId] = keyId
+        self.inputValues[self.keyIdKey] = keyId
     }
     
     /// Sets `token` value in callback response
     /// - Parameter token: Base64 String value of verification
-    public func setVerification(_ token: String) {
-        self.inputValues[self.assertToken] = token
+    public func setAssertion(_ token: String) {
+        self.inputValues[self.tokenKey] = token
     }
     
     /// Sets `clientData` value in callback response
     /// - Parameter clientData: Base64 String value of clientData
     public func setClientData(_ clientData: String) {
-        self.inputValues[self.clientData] = clientData
+        self.inputValues[self.clientDataKey] = clientData
     }
     
     /// Attest the device for iOS14 and above devices
     /// - Throws: `FRDeviceCheckAPIFailure`
     /// - Parameter attestation: Optional Protocol for providing a ``FRAppAttestation`` to implement own attestation
     @available(iOS 14.0, *)
-    public func attest() async throws {
+    open func requestIntegrityToken() async throws {
         do {
-            let result = try await FRAppAttestDomainModal.shared.attest(challenge: challenge)
-            self.setAttestation(result.attestKey)
-            self.setVerification(result.assertKey)
+            let result = try await FRAppAttestDomainModal.shared.requestIntegrityToken(challenge: challenge)
+            self.setAttestation(result.appAttestKey)
+            if let assertkey = result.assertKey {
+                self.setAssertion(assertkey)
+            }
             self.setkeyId(result.keyIdentifier)
             self.setClientData(result.clientDataHash)
+            self.appIntegritykeys = result
         }
         catch {
             FRLog.e("Error: \(error.localizedDescription)")
```

Comment:
When setting the error to the callback, it set to the actual error or the string "ClientDeviceErrors"? For Server to route to the correct outcome, it should set it to "ClientDeviceErrors", and allow developers to customize the error.

The failure will throw to developer, and if they want to have a custom outcome, they can set using FRAppIntegrityCallback.setClientError()

Reply:
this is addressed already.